### ISSUE TITLE
Remapping of one input variable to multiple new ones

### ISSUE THREAD

**Issue by sahahner** [ecmwf/anemoi-models#15]
### Is your feature request related to a problem? Please describe.

Specific problem: We want the model to treat a variable x in degrees in range [0,360] as cos(x) and sin(x) inside the model and map the output back to the representation in degree. 

### Describe the solution you'd like

**Preprocessor**: 
The model should handle the remapped variables, and the training loss must be calculated on the the remapped set of variables. The validation loss should be calculated on the original variable, which will also be output in inference. Therefore, this remapping should be implemented as a preprocessor that allows the remapping of one variable to several new ones. 

**Data Indices:** The preprocessors are the first layers of the model, so the input variables are the same.
After the preprocessors, the set of variables has changed. Therefore, this information needs to be included in the data indices.

### Describe alternatives you've considered

Changing the set of variables when creating the dataset is not an option since the model output needs to contain the variables we are interested in for inference.

### Organisation

ECMWF
